recipes/CatGifsReactHooks/src/Main.purs

```diff
@@ -55,33 +39,34 @@ main = do
 mkCatGifs :: Component {}
 mkCatGifs = do
   component "CatGifs" \_ -> React.do
-    (gifState /\ dispatch) <- useAffReducer Loading reducer
-    useEffectOnce do
-      dispatch MorePlease
-      pure mempty
+    (resetToken /\ reset) <- useResetToken
+    gifState <- toGifState <$> useAff resetToken getRandomCatGif
     let
-      onClick = handler_ (dispatch MorePlease)
+      onClick = handler_ reset
     pure case gifState of
+      Loading -> R.text "Loading..."
       Failure ->
         R.div_
           [ R.text "I could not load a random cat for some reason. "
           , R.button { onClick, children: [ R.text "Try Again!" ] }
           ]
-      Loading -> R.text "Loading..."
       Success url ->
         R.div_
           [ R.button { onClick, style: css { display: "block" }, children: [ R.text "More Please!" ] }
           , R.img { src: url }
           ]
 
-getRandomCatGif :: Aff (Either String String)
+-- | Collapse nested `Maybe`s to our `GifState` type
+toGifState :: Maybe (Maybe String) -> GifState
+toGifState = maybe Loading (maybe Failure Success)
+
+getRandomCatGif :: Aff (Maybe String)
 getRandomCatGif = do
   response <-
     Affjax.get
       ResponseFormat.json
       "https://api.giphy.com/v1/gifs/random?api_key=dc6zaTOxFJmzC&tag=cat"
   pure do
-    { body } <- lmap printError response
-    body' <- decodeJson body
-    data' <- body' .: "data"
-    data' .: "image_url"
+    -- Using `hush` to ignore the possible error messages
+    { body } <- hush response
+    hush (decodeJson body >>= (_ .: "data") >>= (_ .: "image_url"))
```

Comment:
There was also a breaking change with `decodeJson` which now returns an `Either JsonDecodeError` instead of `Either String`. Instead of trying to unify the error types, I opted to just convert it to `Maybe`.

Reply:
Thanks for clarifying this!